src/libponyc/pass/docgen.c

```diff
@@ -548,6 +548,82 @@ static void doc_type_params(docgen_t* docgen, ast_t* t_params,
     fprintf(docgen->type_file, "]");
 }
 
+// Write the expression to the current type file.
+void doc_fprint(docgen_t* docgen, ast_t* ast)
```

Comment:
to me from doc_fprint, its certainly not clear that this is writing out an expression.